internal/jsonpathwriter/jsonpathwriter.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonpathwriter
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/PaesslerAG/jsonpath"
+)
+
+func Print(w io.Writer, path string, obj interface{}) error {
+	jsonString, err := json.Marshal(obj)
+	if err != nil {
+		return err
+	}
+
+	val := interface{}(nil)
+	err = json.Unmarshal(jsonString, &val)
+	if err != nil {
+		return err
+	}
```

Comment:
I have tried to pass obj directly to `jsonpath` but I keep getting errors

Reply:
this seems to be a limitation on json path, it only works with `map[string]interface{}` or `[]interface{}` but not with `interface{}`, weird I know